frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/PipelineRunTaskRunGraph.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { ChartVoronoiContainer } from '@patternfly/react-charts';
+import { GraphEmpty } from '@console/internal/components/graphs/graph-empty';
+import { formatDuration } from '@console/internal/components/utils/datetime';
+import { LoadingInline } from '@console/internal/components/utils';
+import { usePipelineRunTaskRunPoll } from '../hooks';
+import {
+  PipelineMetricsGraphProps,
+  getRangeVectorData,
+  getYaxisValues,
+} from './pipeline-metrics-utils';
+import { LineChart } from './charts/lineChart';
+import { DEFAULT_CHART_HEIGHT } from '../const';
+
+import './PipelineChart.scss';
+
+const PipelineRunTaskRunGraph: React.FC<PipelineMetricsGraphProps> = ({
+  pipeline,
+  timespan,
+  interval,
+  width = 650,
+}) => {
+  const {
+    metadata: { name, namespace },
+  } = pipeline;
+
+  const [runData, runDataError, runDataLoading] = usePipelineRunTaskRunPoll({
+    name,
+    namespace,
+    timespan,
+    delay: interval,
+  });
+
+  if (runDataLoading) {
+    return <LoadingInline />;
+  }
+
+  const pipelineRunResultData = runData?.data?.result ?? [];
+  if (runDataError || pipelineRunResultData.length === 0) {
+    return <GraphEmpty height={DEFAULT_CHART_HEIGHT} />;
+  }
+
+  const pRuns =
+    getRangeVectorData(runData, (r) => {
+      return r.metric.pipelinerun;
+    }) ?? [];
+  const finalObj: { [x: string]: { x: string; y: number }[] } = pRuns.reduce((acc, prun) => {
+    if (!prun) return acc;
+
+    const obj = prun[prun.length - 1];
+    if (obj?.metric?.task && !acc[obj.metric.task]) {
+      acc[obj.metric.task] = [];
+    }
+    acc[obj.metric.task].push(obj);
```

Comment:
If you do have a task, create an array on it then push into it... if you don't have a task, NPE? Or am I misreading this?

Reply:
`metric.task` & `metric.pipelinerun` will be always available in this prometheus  query response, but i agree the above code snippet looks confusing, so i have changed it to conditionally push the obj to accumulator only if the `metric.task` is  present.